packages/battery/android/src/main/java/io/flutter/plugins/battery/BatteryPlugin.java

```diff
@@ -36,18 +39,32 @@ public static void registerWith(PluginRegistry.Registrar registrar) {
   }
 
   BatteryPlugin(PluginRegistry.Registrar registrar) {
-    this.registrar = registrar;
+    this.application = ((Application) registrar.context());
+    application.registerActivityLifecycleCallbacks(lifecycleCallbacks);
   }
 
-  private final PluginRegistry.Registrar registrar;
+  private final Application application;
+  private final IntentFilter chargedFilter = new IntentFilter(Intent.ACTION_BATTERY_CHANGED);
   private BroadcastReceiver chargingStateChangeReceiver;
 
+  private final ActivityLifecycleCallbacks lifecycleCallbacks =
+      new EmptyActivityLifecycleCallbacks() {
+        @Override
+        public void onActivityDestroyed(Activity activity) {
+          application.unregisterActivityLifecycleCallbacks(lifecycleCallbacks);
+          if (chargingStateChangeReceiver != null) {
+            application.unregisterReceiver(chargingStateChangeReceiver);
+            chargingStateChangeReceiver = null;
+          }
+        }
+      };
+
   @Override
   public void onMethodCall(MethodCall call, Result result) {
     if (call.method.equals("getBatteryLevel")) {
       int batteryLevel = getBatteryLevel();
 
-      if (batteryLevel != -1) {
+      if (batteryLevel > 0) {
```

Comment:
Good catch, lets explain the issue in the PR description (Android may return 0 or Integer.MIN_VALUE for errors).

For future PRs - probably better to separate different bugs to different PRs (easier when going through code history, and easier to review).